pkg/controller/controller.go

```diff
@@ -478,19 +504,48 @@ func (c *Controller) runAction(ctx context.Context, as *crv1alpha1.ActionSet, aI
 			reason := fmt.Sprintf("ActionSetFailed Action: %s", action.Name)
 			msg := fmt.Sprintf("Failed to update Output Artifacts: %#v:", artTpls)
 			c.logAndErrorEvent(ctx, msg, reason, aErr, as, bp)
+			c.executeDeferPhase(deferPhase, tp, bp, action.Name, ns, aIDX)
 			return nil
 		}
 		if err != nil {
 			reason := fmt.Sprintf("ActionSetFailed Action: %s", action.Name)
 			msg := "Failed to render output artifacts"
 			c.logAndErrorEvent(ctx, msg, reason, err, as, bp)
+			c.executeDeferPhase(deferPhase, tp, bp, action.Name, ns, aIDX)
 			return nil
 		}
+
+		c.executeDeferPhase(deferPhase, tp, bp, action.Name, ns, aIDX)
```

Comment:
If the defer phase has an output and is used in the output artifact, where/how do we add it?